addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/handler/LoxoneMiniserverHandler.java

```diff
@@ -0,0 +1,581 @@
+                if (command instanceof PercentType) {
+                    jalousie.moveToPosition(((PercentType) command).doubleValue() / 100);
+                } else if (command instanceof UpDownType) {
+                    if ((UpDownType) command == UpDownType.UP) {
+                        jalousie.fullUp();
+                    } else {
+                        jalousie.fullDown();
+                    }
+                } else if (command instanceof StopMoveType) {
+                    if ((StopMoveType) command == StopMoveType.STOP) {
+                        jalousie.stop();
+                    }
+                }
+                return;
+            }
+
+            if (control instanceof LxControlLightController) {
+                LxControlLightController controller = (LxControlLightController) control;
+                if (command instanceof OnOffType) {
+                    if ((OnOffType) command == OnOffType.ON) {
+                        controller.allOn();
+                    } else {
+                        controller.allOff();
+                    }
+                } else if (command instanceof UpDownType) {
+                    if ((UpDownType) command == UpDownType.UP) {
+                        controller.nextScene();
+                    } else {
+                        controller.previousScene();
+                    }
+                } else if (command instanceof DecimalType) {
+                    controller.setScene(((DecimalType) command).intValue());
+                }
+                return;
+            }
+
+            if (control instanceof LxControlRadio) {
+                LxControlRadio radio = (LxControlRadio) control;
+                if (command instanceof OnOffType) {
+                    if ((OnOffType) command == OnOffType.OFF) {
+                        radio.setOutput(0);
+                    }
+                } else if (command instanceof DecimalType) {
+                    radio.setOutput(((DecimalType) command).intValue());
+                }
+                return;
+            }
+
+            logger.debug("Incompatible operation on control {}", control.getUuid());
+
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        logger.debug("Channel linked: {}", channelUID.getAsString());
+        LxControl control = getControlFromChannelUID(channelUID);
+        if (control != null) {
+            updateChannelStates(control);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.trace("Initializing thing");
+        switchTypeId = addNewChannelType("switch", "Switch", "Switch", "Loxone Switch", false);
```

Comment:
I will move them to the xml, especially that we plan to get rid of dynamic channel creation here at all in the future. But a bit of reasoning for having them here is of esthetic nature - since I create other channels dynamically, I liked creating all of them in the same way from within the code.